src/iota/signing.c

```diff
@@ -23,31 +23,29 @@ static uint16_t sign_signature(SIGNATURE_BLOCK *pBlock,
                                uint32_t *bip32_signing_path,
                                API_INPUT_BIP32_INDEX *input_bip32_index)
 {
-    cx_ecfp_private_key_t pk;
-    cx_ecfp_public_key_t pub;
 
     // overwrite bip32_signing_path
     bip32_signing_path[BIP32_ADDRESS_INDEX] = input_bip32_index->bip32_index;
     bip32_signing_path[BIP32_CHANGE_INDEX] = input_bip32_index->bip32_change;
 
-    uint32_t signature_length = 0;
+    size_t signature_length = CX_SHA512_SIZE;
 
-    uint8_t ret = 0;
-    // create key pair and convert pub key to bytes
-    ret = ed25519_get_key_pair(bip32_signing_path, BIP32_PATH_LEN, &pk, &pub);
-    ret = ret && ed25519_sign(&pk, essence_hash, BLAKE2B_SIZE_BYTES,
-                              pBlock->signature, &signature_length);
+    MUST(bip32_derive_with_seed_eddsa_sign_hash_256(
+             HDW_ED25519_SLIP10, CX_CURVE_Ed25519, bip32_signing_path,
+             BIP32_PATH_LEN, CX_SHA512, essence_hash, BLAKE2B_SIZE_BYTES,
+             pBlock->signature, &signature_length, NULL, 0) == CX_OK);
```

Comment:
```suggestion
    MUST(bip32_derive_eddsa_sign_hash_256(
             HDW_ED25519_SLIP10, CX_CURVE_Ed25519, bip32_signing_path,
             BIP32_PATH_LEN, CX_SHA512, essence_hash, BLAKE2B_SIZE_BYTES,
             pBlock->signature, &signature_length) == CX_OK);
```

Reply:
There is no parameter to pass HDW_ED25519_SLIP10. Calculated values differ then.